airbyte-webapp/src/components/connection/CatalogTree/next/StreamDetailsPanel/StreamPanelHeader/StreamPanelHeader.tsx

```diff
@@ -19,17 +19,18 @@ interface StreamPanelHeaderProps {
   stream?: AirbyteStream;
 }
 
-interface SomethingProps {
+interface StreamPropertyProps {
   messageId: string;
   value?: string | ReactNode;
+  "data-testid"?: string;
 }
 
-export const StreamProperty: React.FC<SomethingProps> = ({ messageId, value }) => (
-  <span>
-    <Text size="sm" className={styles.streamPropLabel}>
+export const StreamProperty: React.FC<StreamPropertyProps> = ({ messageId, value, "data-testid": testId }) => (
```

Comment:
Why not just have the prop be `testId`?

Reply:
To prevent people from guessing if it's `testId` or data-testid. Some components extend the HTML element props so they accept it, other's don't.